MainWindow.cpp

```diff
@@ -181,27 +183,52 @@ void MainWindow::openFile(QString fName) {
   QFileInfo fileInfo(fName);
   const QString suffix = fileInfo.suffix();
 
+  buffers::Project *loadedProject = nullptr;
   if (suffix == "gm81" || suffix == "gmk" || suffix == "gm6" || suffix == "gmd") {
-    project = gmk::LoadGMK(fName.toStdString());
+    loadedProject = gmk::LoadGMK(fName.toStdString());
   } else if (suffix == "gmx") {
-    project = gmx::LoadGMX(fName.toStdString());
+    loadedProject = gmx::LoadGMX(fName.toStdString());
   } else if (suffix == "yyp") {
-    project = yyp::LoadYYP(fName.toStdString());
+    loadedProject = yyp::LoadYYP(fName.toStdString());
   }
 
-  if (!project) {
+  if (!loadedProject) {
     QMessageBox::warning(this, tr("Failed To Open Project"), tr("There was a problem loading the project: ") + fName,
                          QMessageBox::Ok);
     return;
   }
 
   MainWindow::setWindowTitle(fileInfo.fileName() + " - ENIGMA");
   recentFiles->prependFile(fName);
+  openProject(loadedProject);
+}
+
+void MainWindow::openNewProject() {
+  MainWindow::setWindowTitle(tr("<new game> - ENIGMA"));
+  auto newProject = new buffers::Project();

```

Comment:
1. Try to put `*` after `auto` anywhere you know it's a pointer.
2. Why isn't this a `unique_ptr`? In general, please don't say `new` anywhere in the codebase.

Reply:
Sorry about (1) that was a mistake, I intended to do that. As far as (2) it is a unique_ptr now on MainWindow, but I haven't switched libegm to that yet because I didn't want to make conflicts (gmk/gmx have a unique_ptr they are just releasing it before return; don't ask me why I did that it was silly of me to create this technical debt, I know), I am planning to do this soon. Regardless, that's why the `MainWindow::openProject` takes a raw pointer. However, I can make a commit that fixes half of it, give me a second.

Reply:
Alright: 1a99871cb3b7ec0b44627a7b4a79c9b41093ffbb

I'll get to having all the libegm load methods returning a unique_ptr soon.